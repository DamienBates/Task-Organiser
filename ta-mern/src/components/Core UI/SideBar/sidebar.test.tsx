import SideBar from "./SideBar"
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import renderer from "react-test-renderer"

configure({ adapter: new Adapter() });

describe("sidebar tests", function () {
    test("navlinks display", () => {
        expect(SideBar).toBeDefined();
    });

    test('should render', () => {
        const component = renderer
            .create(
                <MemoryRouter>
                    <SideBar />
                </MemoryRouter>
            )
            .toJSON();

        expect(component).toMatchSnapshot();
    });
})