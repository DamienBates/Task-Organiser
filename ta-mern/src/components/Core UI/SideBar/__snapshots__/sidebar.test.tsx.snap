// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar tests should render 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked css-1vaj88k-MuiDrawer-docked"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "100vh",
          "justifyContent": "center",
          "width": "45px",
        }
      }
    >
      <div
        className="MuiBox-root css-2ishhi"
      >
        <a
          href="/"
          onClick={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="MuiBox-root css-0"
            onClick={[Function]}
            style={
              Object {
                "color": "#828282",
                "textDecoration": "none",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
        </a>
      </div>
      <div
        className="MuiBox-root css-2ishhi"
      >
        <a
          href="/TaskList"
          onClick={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="MuiBox-root css-0"
            onClick={[Function]}
            style={
              Object {
                "color": "#828282",
                "textDecoration": "none",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CreateIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </div>
        </a>
      </div>
    </li>
  </div>
</div>
`;
